### PR TITLE
ignore composer lock.

### PR DESCRIPTION
**Pull request by abacaphiliac**
this is a lib, right? composer.lock shouldn't be necessary.

### CODE REVIEW

**Comment by abacaphiliac** [comment]
@jeremygiberson ahh, there we go. build passes. so, i had to version down composer and symfony in order to make them compatible again. looks like the method signature for \Symfony\Component\Console\Application::renderException includes type-hints in ^3.0, but \Composer\Console\Application::renderException didn't get the message yet in dev-master. i'm looking for advice on this one.

**Comment by abacaphiliac** [comment]
sorry about the thrashing. ok, so someone renamed \Composer\Command\Command to \Composer\Command\BaseCommand 6 days ago. one of your Helpers relies on the old name, so i've updated the helper. alternatively, we can version down composer (so that it's more stable), but that means we have to version down symfony : (

**Comment by abacaphiliac** [comment]
here's the bc-break in composer: https://github.com/composer/composer/commit/692a3ed300daaede8f4fcf4176eb8ca2048c6d87

**Comment by jeremygiberson** [comment]
@abacaphiliac Thanks for the PR. This is a library but it is also a tool. Generally this package will be installed separately from your actual projects. 

See https://github.com/composer/composer for precedence.

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,5 @@
 .idea/
+build/
 vendor/
 testdir/
+composer.lock
```
Comment (jeremygiberson) [inline .gitignore]: please remove this. and bring back composer lock. I think you'll need to do a composer update symfony/yaml though as their commit history has been changed and the locked hash is no longer available.  With these changes I can accept the PR.

**Comment by abacaphiliac** [comment]
@jeremygiberson gotcha. ok i added composer.lock back to this request, meaning this request is just to update dependency versions. does installation as a tool mean that the dependencies of this tool are isolated from the application it is being installed into? or can they conflict? for installation as a tool, we might want to generate a phar.

**Comment by jeremygiberson** [comment]
@abacaphiliac 

> does installation as a tool mean that the dependencies of this tool are isolated from the application it is being installed into?

The tool and dependencies of the tool are isolated from the application because the application never knows about the tool. Its completely independent of any project you are working on. 

See [robs blog](https://akrabat.com/global-installation-of-php-tools-with-composer/) -- notice how global installs generate their own vendor & autoload stuff in your home directory. Global installs don't impact your per-project composer ecosystem.

However, as currently documented as a global composer install, its dependencies could conflict with other packages you have globally installed (because all global packages share the same homedir ecosystem). Which could be problematic for people that already have other globally installed tools with lib dependency versions that conflict with quickstrap. 

I'll probably be up for creating a phar for distribution that can be installed independently of composer  entirely. But I believe that would be part of the release pipeline and not contained within the gitrepo. Here is an [interesting project](https://github.com/clue/phar-composer) that might be useful when we get to that point (people are using this tool and want an alternative installation method than global composer require).

File: .travis.yml [hunk 2]
```diff
@@ -11,5 +11,5 @@ before_script:
   - composer install
 
 script:
-  - phpunit --coverage-clover build/logs/clover.xml
+  - vendor/bin/phpunit --coverage-clover build/logs/clover.xml --coverage-text
```
Comment (jeremygiberson) [inline .travis.yml]: Sorry I didn't catch this earlier but please remove `--coverage-text`. The coverage being generated is for use with code-climate which parses the default output format.

Comment (abacaphiliac) [inline .travis.yml]: good catch. my bad.

File: composer.json [hunk 3]
```diff
@@ -9,14 +9,14 @@
...
-    "minimum-stability": "dev",
+    "minimum-stability": "stable",
     "require": {
         "php": ">=5.6",
         "symfony/console": "^3.0",
         "symfony/process": "^3.0",
         "composer/composer": "dev-master",
         "symfony/event-dispatcher": "^3.0",
-        "servo/fluidxml": "^1.13"
+        "servo/fluidxml": "1.13"
```
Comment (abacaphiliac) [inline composer.json:19]: when i updated to ^1.2 we got a truckload of test failures. i thought it best to lock that down temporarily.

**Comment by jeremygiberson** [comment]
@abacaphiliac thanks for your contribution!

**Pull request merged**